File: Day334/ImageSmoother.java
class Solution {
    public int[][] imageSmoother(int[][] img) {
        int dir[][] ={{-1,-1}, {-1,0}, {-1,1}, {0,-1}, {0,0}, {0,1}, {1,-1},{1,1}, {1,0}};
        int answer[][] = new int[img.length][img[0].length];
        for (int i=0; i<img.length; i++) {
            for ( int j=0; j<img[0].length; j++) {
                int temp = 0;
                int count = 0;
                for (int v[] :  dir ){
                    int x = i + v[0];
                    int y =  j+v[1];
                    if (x>=0 && y>=0 && x<img.length && y<img[0].length) {
                        temp += img[x][y];
                        count++;
                    }
                }
                answer[i][j] = temp/count;
            }
        }
            return answer;
    }
}

File: Day334/ImageSmoother.go
func imageSmoother(img [][]int) [][]int {
    dir := [][]int{{-1,-1}, {-1,0}, {-1,1}, {0,-1}, {0,0}, {0,1}, {1,-1},{1,1}, {1,0}}
    answer := make([][]int, len(img))
    for i:=0; i<len(img); i++ {
        answer[i] = make([]int, len(img[0]))
        for j:=0; j<len(img[0]); j++ {
            temp := 0
            count := 0
            for _,v := range dir {
                x, y := i + v[0], j+v[1]
                if x>=0 && y>=0 && x<len(img) && y<len(img[0]) {
                    temp += img[x][y]
                    count++
                }
            }
            answer[i][j] = temp/count
        }
    }
        return answer
}